components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -360,7 +360,7 @@ public IdentityProvider getResidentIdP(String tenantDomain)
         }
         propertiesList = new ArrayList<Property>();
 
-        Property idPEntityIdProp = resolveFedAuthnProperty(getOIDCResidentIdPEntityId(), oidcFedAuthn,
+        Property idPEntityIdProp = resolveFedAuthnProperty(oauth2TokenEPUrl, oidcFedAuthn,
```

Comment:
Ideally, the fix should be done in the private `getOIDCResidentIdPEntityId` method. The URL should be resolved by calling `resolveAbsoluteURL` within the method.

Reply:
Please check how it would affect the other usages of the method as well.

Reply:
If we fix it in a private method, it will be persisted in the DB. So when the `tenant-qualified URLs` is disabled, still the value will have the tenanted path. Is this expected?

Reply:
If thats the case, we can introduce another variable similar to the other property variables and resolve the URL for the returned value of this method. 
This would be better because there could be instances where the `idpEntityId` is not always equivalent to the OAuth2 token endpoint value. In the default product, the Entity ID is a customizable value.